services/api/src/lib/scraper/booking_apify.py

```
@@ -61,89 +76,120 @@ def generate_request(self, user_request: GeneratedRequirement):
+        results = await asyncio.gather(*tasks)
+
+        properties: list[BookingApifyResponse] = []
+
+        # Process all results instead of just the first one
+        for result in results:
+            if result and "defaultDatasetId" in result:
+                # Get dataset items in a non-blocking way
+                dataset_id = result["defaultDatasetId"]
+
+                items = await loop.run_in_executor(
+                    None, lambda: list(self.client.dataset(dataset_id).iterate_items())
```

Ruff (0.8.2)

Function definition does not bind loop variable `dataset_id`

(B023)